src/odb/src/zutil/util.cpp

```diff
@@ -279,8 +279,29 @@ bool hasOneSiteMaster(dbDatabase* db)
 {
   for (dbLib* lib : db->getLibs()) {
     for (dbMaster* master : lib->getMasters()) {
+      if (master->isBlock() || master->isPad() || master->isCover()) {
+        continue;
+      }
+
+      // Ignore IO corner cells
+      dbMasterType type = master->getType();
+      if (type == dbMasterType::ENDCAP_TOPLEFT
+          || type == dbMasterType::ENDCAP_TOPRIGHT
+          || type == dbMasterType::ENDCAP_BOTTOMLEFT
+          || type == dbMasterType::ENDCAP_BOTTOMRIGHT) {
+        continue;
```

Comment:
Use `mater->getType().isEndCap()` (there are quite a few more types for endcaps)

Reply:
only the 4 listed refer to PAD corners, the remainder refer to standard cell end caps. It's a dumb LEF naming thing, but isEndCap captures both standard cells (which we want to check) and IO corners (which we dont)